File: src/main/java/com/vc/easy/L242.scala
package com.vc.easy

object L242 {
  def isAnagram(s: String, t: String): Boolean = {
    import scala.collection.mutable
    val map = new mutable.HashMap[Char, Int]()
    s.foreach(ch => {
      map.put(ch, map.getOrElse(ch, 0) + 1)
    })
    t.foreach(ch => {
      if(map.contains(ch) && map(ch) == 1) map.remove(ch)
      else map.put(ch, map.getOrElse(ch, 0) - 1)
    })
    map.isEmpty
  }
}

File: src/main/java/com/vc/easy/L242.java
package com.vc.easy;

import java.util.HashMap;

class L242 {
    public boolean isAnagram(String s, String t) {
        HashMap<Character, Integer> map = new HashMap<>();

        for(int i = 0; i < s.length(); i++) {
            map.put(s.charAt(i), map.getOrDefault(s.charAt(i), 0) + 1);
        }

        for(int i = 0; i < t.length(); i++) {
            char ch = t.charAt(i);
            if(map.getOrDefault(ch, 0) == 0) return false;
            map.put(ch, map.get(ch) - 1);
            if(map.get(ch) == 0) map.remove(ch);
        }

        return map.size() == 0;
    }
}
